artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java

```diff
@@ -268,6 +271,8 @@ private boolean internalStop(boolean ioCriticalError, boolean sendFailover) thro
             idGenerator.persistCurrentID();
       }
 
+      deletingLargeMessageTasks.await(30, TimeUnit.SECONDS);
```

Comment:
-1 This isnt a good idea. On case of split brain with network pinger and network is detected split we need the isolated to shutdown to occur immediately without any blocking.

Reply:
I see your point but it normally should not block and 10 lines forward I see the same code:

> latch.await(30, TimeUnit.SECONDS);

Reply:
that doesnt mean its something we would want to proliferate. Just because a window is broken, doesnt mean its ok to just go inside... avoid the broken glass...